heim-runtime/src/linux.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(target_os = "linux")]
+
+/// Linux-specific extension for configuring custom paths for Procfs
+/// and Sysfs.
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+use std::path::{Path, PathBuf};
+
+/// Holds global state for custom paths. Static's leveraging this type
+/// can only be set ONCE per binary run.
+type RootLock = OnceCell<PathBuf>;
+
+static PROCFS_ROOT: RootLock = OnceCell::new();
+static SYSFS_ROOT: RootLock = OnceCell::new();
+
+/// Instantiates the OnceCell holding PROCFS_ROOT in the case that it hasn't
+/// already been instantiated.
+///
+/// This can only be set ONCE per binary run.
+pub fn set_procfs_root<T: Into<Cow<'static, Path>>>(root: T) {
+    let root = root.into().into_owned();
+    let _ = PROCFS_ROOT.get_or_init(|| root);
+}
+
+/// Returns the static `Path` value of a configured PROCFS_ROOT.
+///
+/// Returns default of '/proc' if uninitialized
```

Comment:
More precise would be
```
If uninitialized, initializes the PROCFS_ROOT with the default path - `/proc`
```

This wording avoids confusion regarding whether value can (it can't!) be set after it's read. Also, we might want to explicitly document this - that you must set before the first read.

The same applies to sysfs.